waku/waku_archive/driver/postgres_driver/postgres_driver.nim

```diff
@@ -312,8 +318,13 @@ method deleteOldestMessagesNotWithinLimit*(
 method close*(s: PostgresDriver):
               Future[ArchiveDriverResult[void]] {.async.} =
   ## Close the database connection
-  let result = await s.connPool.close()
-  return result
+  (await s.writeConnPool.close()).isOkOr:
+    return err("error closing write pool: " & $error)
+
+  (await s.readConnPool.close()).isOkOr:
+    return err("error closing read pool: " & $error)
+
+  return ok()
 
```

Comment:
If `s.writeConnPool.close()` fails, shouldn't we still attempt to run `s.readConnPool.close()` before returning?